tests/unit/infrastructure/app.test.ts

```diff
@@ -0,0 +1,59 @@
+import supertest from 'supertest';
+import { app } from '../../../src/infrastructure/api';
+import { DataAccess } from '../../../src/utils/data-access';
+
+describe('API', () => {
+  afterEach(() => {
+    jest.resetAllMocks().restoreAllMocks();
+  });
+
+  describe('POST /', () => {
+    test('should return status code of 204 when inserting a valid trailer registration', async () => {
+      const payload = {
+        vin: 'ABC1321234566',
+        make: 'big truck',
+        trn: 'AB123AD',
+        certificateExpiryDate: '2021-12-12',
+        certificateIssueDate: '2021-01-01',
+      };
+
+      const dataInstanceSpy = jest.spyOn(DataAccess, 'getInstance');
+
+      dataInstanceSpy.mockReturnValue(({
+        getByVinOrChassisWithMake: () => null,
+        upsertTrailerRegistration: () => {},
+      } as unknown) as DataAccess);
+      return supertest(app).post('/v1/trailers').send(payload).expect(204);
+    });
+
+    test('should return status code of 400 when inserting an invalid trailer registration', async () => {
+      const payload = {
+        vin: 'ABC1321234566',
+        make: 'big truck',
+        certificateExpiryDate: '2021-12-12',
+        certificateIssueDate: '2021-01-01',
+      };
+
+      const result = await supertest(app).post('/v1/trailers').send(payload);
+      expect(result.status).toEqual(400);
+      expect(result.text).toEqual('"trn" is required');
+    });
+  });
+
+  describe('PUT /deregister/:trn', () => {
+    test('should return status code 204 if an existing trailer for the provided trn is not found', async () => {
```

Comment:
Should we have a test to ensure we get a 204 if an existing trailer for the provided trn is found ?

Reply:
more tests for dregister are defined in `deregister-trailer.test.ts`